src/components/GrainResults.js

```diff
@@ -2,53 +2,74 @@ import React, { useEffect, useState } from 'react';
 import '../css/GrainResults.css';
 import Search from './Search'
 import Grain from './Grain'
-import data from '../mockData'
+import { useQuery, gql } from '@apollo/client';
 
-const GrainResults = () => {
+const GET_FARMERS = gql`
+  query {
+    allFarmers {
+      name
+      id
+      region
+      grains {
+        name
+        id
+        moisture
+        fallingNumber
+        protein
+        testWeight
+        farmersNotes
+        farmerId
+      }
+    }
+  }
+`;
 
-  // need to fetch all grains from server and setState with setGrains
-  // need to fetch all farms with just .id, .name & .region
+const GrainResults = () => {
 
-  const [grains, setGrains] = useState([])
-  const [farms, setFarms] = useState([])
   const [search, setSearch] = useState('')
   const [filtered, setFiltered] = useState([])
 
-  const grainCards = grains.map(grain => {
-    let farm = farms.find(farm => grain.farm_id === farm.id)
-    return (
-      <Grain key={grain.id} grain={grain} farm={farm}/>
-      )
-  })
+  const { loading, error, data } = useQuery(GET_FARMERS);
+
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error :(</p>;
```

Comment:
Could we display a more helpful error message to the user here?

Reply:
Absolutely!  We had this in three files and I have updated each to "Error: Unable to connect to the server, please try again later." on my current working branch, if that's okay?

Reply:
Yeah that makes sense as long as there aren't any other cases to consider like authorization errors.